gemd/units/impl.py

```diff
@@ -3,70 +3,132 @@
 
 from pint import UnitRegistry, Unit, register_unit_format, Quantity
 from pint.compat import tokenizer
-from tokenize import NAME, NUMBER, OP
+from tokenize import NAME, NUMBER, OP, Token, ERRORTOKEN
 # alias the error that is thrown when units are incompatible
 # this helps to isolate the dependence on pint
 from pint.errors import DimensionalityError as IncompatibleUnitsError  # noqa Import
 from pint.errors import UndefinedUnitError, DefinitionSyntaxError  # noqa Import
 
 import functools
 import pkg_resources
-from typing import Union
+from typing import Union, List, Tuple
 
 # use the default unit registry for now
 DEFAULT_FILE = pkg_resources.resource_filename("gemd.units", "citrine_en.txt")
 _ALLOWED_OPERATORS = {".", "+", "-", "*", "/", "//", "^", "**", "(", ")"}
 
 
+def _space_after_minus_preprocessor(input_string: str) -> str:
+    """A preprocessor that protects against a pint < 0.21 bug."""
+    return re.sub(r"(?<=-)\s+(?=\d)", "", input_string)
```

Comment:
Inconsistent behavior between 0.20 & 0.21 for `m ** - 1` (vs. `m ** -1`).  Older version threw a parsing error, so this corrects for that bug by avoiding the messy condition.

Reply:
I don't think that buys us much unless it's compiled outside the function.

Reply:
Right. Any reason it can't be moved outside the function?